decoder/src/elf2table/mod.rs

```diff
@@ -89,28 +90,68 @@ pub fn parse_impl(elf: &[u8], check_version: bool) -> Result<Option<Table>, anyh
             continue;
         }
 
-        if entry.section_index() == Some(defmt_shndx) {
+        if entry.section_index() == Some(defmt_section.index()) {
             let sym = symbol::Symbol::demangle(name)?;
             match sym.tag() {
-                symbol::SymbolTag::Defmt(tag) => {
-                    if tag == Tag::Timestamp {
-                        if timestamp.is_some() {
-                            bail!("multiple timestamp format specifications found");
-                        }
+                symbol::SymbolTag::Defmt(Tag::Timestamp) => {
+                    if timestamp.is_some() {
+                        bail!("multiple timestamp format specifications found");
+                    }
 
-                        timestamp = Some(TableEntry::new(
-                            StringEntry::new(tag, sym.data().to_string()),
-                            name.to_string(),
-                        ));
-                    } else {
-                        map.insert(
-                            entry.address() as usize,
-                            TableEntry::new(
-                                StringEntry::new(tag, sym.data().to_string()),
-                                name.to_string(),
-                            ),
+                    timestamp = Some(TableEntry::new(
+                        StringEntry::new(Tag::Timestamp, sym.data().to_string()),
+                        name.to_string(),
+                    ));
+                }
+                symbol::SymbolTag::Defmt(Tag::BitflagsValue) => {
+                    // Bitflags values always occupy 128 bits / 16 bytes.
+                    if entry.size() != 16 {
+                        bail!(
+                            "bitflags value does not occupy 16 bytes (symbol `{}`)",
+                            name
                         );
                     }
+
+                    let defmt_data = defmt_section.data()?;
```

Comment:
ah, I guess this is where the read the data stored in the `static` produce by `defmt::bitflags`. could we *not* put the value in the static but instead put it in its symbol name?

Reply:
No, since that would require evaluating arbitrary constant expressions during proc macro evaluation

Reply:
> No, since that would require evaluating arbitrary constant expressions during proc macro evaluation

right, that's what I feared. ~~then, would it be possible to put the flag values in a *different* output linker section (e.g. `.defmt.bitflags`) so that each flag doesn't waste 15 (16? (+)) additional bytes of the main `.defmt` linker section?

(+) do we ever use the address of the `static`s defined in the `assert` function?

I guess before that question, the more relevant question is: could bitflag flags exhaust the 16-bit `.defmt` address space? You need ~4000 flags for that and you can have more than one flag per `bitflags!` invocation.~~

I see that you put these values after the format string indices in the linker script, which is roughly equivalent to having a separate section, so they won't consume the 16-bit `.defmt` address space. Never mind this comment then.